imageregistry/v1/types.go

```diff
@@ -90,6 +98,10 @@ type ImageRegistrySpec struct {
 	// affinity is a group of node affinity scheduling rules for the image registry pod(s).
 	// +optional
 	Affinity *corev1.Affinity `json:"affinity,omitempty"`
+	// storageManagementState indicates if the operator manages the underlying storage unit.
+	// +optional
+	// +kubebuilder:validation:Pattern=`^(Managed|Unmanaged)$`
+	StorageManagementState string `json:"storageManagementState,omitempty"`
```

Comment:
why not put this inside the ImageRegistryConfigStorage struct?

Reply:
Sounds good. Updated accordingly. PTAL.